### PR TITLE
Add novel method to predict modality task

### PR DESCRIPTION
**Pull request by KaiWaldrant**
## Issue ticket number and link
Closes #xxxx (Replace xxxx with the GitHub issue number)

## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [x] CI Tests succeed and look good!

### CODE REVIEW

**Comment by KaiWaldrant** [comment]
* A lot of hardcoded dimensions that are unclear on where they are coming from wich results in uneven tensors. This means there is no model output.

File: src/tasks/predict_modality/methods/novel/train/script.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+## VIASH END
+
+
+sys.path.append(meta['resources_dir'])
+from helper_functions import train_and_valid, lsiTransformer, ModalityMatchingDataset
+from helper_functions import ModelRegressionAtac2Gex, ModelRegressionAdt2Gex, ModelRegressionGex2Adt, ModelRegressionGex2Atac
+
+print('Load data', flush=True)
+
+# TODO: something went really wrong while refactoring this component,
```
Comment (rcannood) [inline src/tasks/predict_modality/methods/novel/train/script.py:39]: Remove these comments

File: src/tasks/predict_modality/methods/novel/run/main.nf [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  main:
+  output_ch = input_ch
+    | novel_train.run(
+      fromState: ["input_train_mod1", "input_train_mod2"],
+      toState: ["input_model": "output"]
+    )
+    | novel_predict.run(
+      fromState: ["input_test_mod1", "input_model"],
+      toState: ["output": "output"]
+    )
```
Comment (rcannood) [inline src/tasks/predict_modality/methods/novel/run/main.nf:12]: do we need to add `output_transform` to this?

File: src/tasks/predict_modality/methods/novel/train/script.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+# because the train script is not supposed to get access to the test data.
+# see this version of the code to start:
+# https://github.com/openproblems-bio/neurips2021_multimodal_topmethods/blob/dc7bd58dacbe804dcc7be047531d795b1b04741e/src/predict_modality/methods/novel/train/script.py#L27-L28
+
+
```
Comment (rcannood) [inline src/tasks/predict_modality/methods/novel/train/script.py:44]: Can remove these comments now

```suggestion
```

File: src/common/library.bib [hunk 4]
```diff
@@ -1066,6 +1066,23 @@ @article{pliner2019supervised
...
+}
+
+
```
Comment (rcannood) [inline src/common/library.bib:1085]: this publication is already in the `library.bib` under `lance2022multimodal`.

```suggestion
```

File: src/tasks/predict_modality/methods/novel/run/config.vsh.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+__merge__: ../../../api/comp_method_train.yaml
+functionality:
+  name: novel
+  info:
+    label: Novel
+    summary: A training method using encoder-decoder MLP model
+    description: This method trains an encoder-decoder MLP model with one output neuron per component in the target. As an input, the encoders use representations obtained from ATAC and GEX data via LSI transform and raw ADT data. The hyperparameters of the models were found via broad hyperparameter search using the Optuna framework.
+    documentation_url: https://github.com/openproblems-bio/neurips2021_multimodal_topmethods/tree/main/src/predict_modality/methods/novel#readme
+    repository_url: https://github.com/openproblems-bio/neurips2021_multimodal_topmethods/tree/main/src/predict_modality/methods/novel
+    reference: pmlr-v176-lance22a
```
Comment (rcannood) [inline src/tasks/predict_modality/methods/novel/run/config.vsh.yaml:10]: this publication is already in the `library.bib` under `lance2022multimodal`.

```suggestion
    reference: lance2022multimodal
```

**Comment by rcannood** [comment]
Could you merge main into this branch? 😊

**Review by rcannood — COMMENTED** [review]
minor suggestion

**Pull request merged**